dashboard/app/models/channel_token.rb

```diff
@@ -87,14 +87,12 @@ def self.find_channel_token(level, user_storage_id, script_id)
   def self.create_channel(ip, project, data: {}, src: nil, type: nil, remix_parent_id: nil, standalone: true, level: nil)
     if src
       data = project.get(src)
-      data['name'] = "Remix: #{data['name']}"
-      data['hidden'] = false
-      data['frozen'] = false
+      data.merge!(name: "Remix: #{data['name']}", hidden: false, frozen: false)
     end
 
     timestamp = Time.now
     project.create(
-      data.merge('createdAt' => timestamp, 'updatedAt' => timestamp),
+      data.merge(createdAt: timestamp, updatedAt: timestamp),
```

Comment:
If data doesn't get set in the `if src` block above, then we'd be merging with whatever `data` object was passed in. Could that one have string keys? Would that be OK in this case because create() favors symbols?

Reply:
It looks like `ChannelToken.create_channel` is called on when a project is remixed, when a new project [is created](https://github.com/code-dot-org/code-dot-org/blob/d42c6bf7f87863f1c24374e04f42fac3061ea1aa/dashboard/app/controllers/projects_controller.rb#L302), or when a project [is exported](https://github.com/code-dot-org/code-dot-org/blob/d42c6bf7f87863f1c24374e04f42fac3061ea1aa/dashboard/app/controllers/projects_controller.rb#L523). For the last two cases, `data` is assigned `initial_data` defined [here](https://github.com/code-dot-org/code-dot-org/blob/d42c6bf7f87863f1c24374e04f42fac3061ea1aa/dashboard/app/controllers/projects_controller.rb#L612) and the keys are symbols.

Reply:
Ahh, sweet! Thank you for checking that and the thorough explanation!